### DISCUSSION TITLE
Linking multiple sets of data together

### DISCUSSION THREAD

**Discussion by cmartino88** [Budibase/budibase#7701] [Help]
I'm attempting to build what I believe to be a relatively simple app however I have 0 experience in databases so I assume I'm missing something somewhere. I have some skill in C++ but I don't think that translates here. Anyway what I'm trying to build is a list of customers. Customers can be added and edited by anyone but only deleted by an admin, haven't worked that part out but that's of least concern.

My biggest issue is we have a list of serialized products, each one can only exist once and will be assigned to a customer. Some customers have divisions and the product would be assigned to that instead of the main customer. I successfully created a table of customers that's searchable, I sucessfully created a table of products. No matter which customer I click though those products show for every customer, I cant figure out how to make them assigned to one customer and one only. I also can't figure out how to make it so the serial number can't be duplicated elsewhere when adding a new product I can't find a way to validate it....

I'm sorry if this seems dumb or simple, perhaps longwinded but I am completely lost here...

Thank you!

**Comment by melohagan** [comment]
Hi @cmartino88 

There's an example here of displaying Products that seems pretty close to what you're trying to do: https://docs.budibase.com/docs/fetching-data#displaying-relational-data

To avoid duplicate serial numbers, you'll need to do some sort of lookup. There's a discussion on that here: https://github.com/Budibase/budibase/discussions/6294
